File: WineriesApp/src/WineriesApp.UI/src/models/Filters/WinesFilter.ts
import { WineType } from "../../enums/WineType";

export class WinesFilter {
    public SearchTerm: string;
        
    public Ratings: number[];

    public Types: WineType[];

    public constructor(searchTerm: string = '', ratings: number[] = [], types: WineType[] = []) {
        this.SearchTerm = searchTerm;
        this.Ratings = ratings;
        this.Types = types;
    }
}

File: WineriesApp/src/WineriesApp.Services/Models/Filters/WinesFilter.cs
using WineriesApp.DataContext.Enums;

namespace WineriesApp.Services.Models.Filters
{
    public class WinesFilter : BaseFilter
    {
        public string? SearchTerm { get; set; }
        
        public double[] Ratings { get; set; } = Array.Empty<double>();

        public WineType[] Types { get; set; } = Array.Empty<WineType>();
    }
}